src/System.Text.Json/tests/Utf8JsonReaderTests.cs

```diff
@@ -629,30 +629,40 @@ public static void SingleJsonValue(string jsonString, string expectedString)
+        [InlineData("false", 4, 0, false)]  // "fals"
+        [InlineData("   {\"a\u6F22\u5B57ge\":30}   ", 16, 16, false)] // "   {\"a\u6F22\u5B57ge\":"
+        [InlineData("{\"n\u6F22\u5B57ame\":\"A\u6F22\u5B57hson\"}", 15, 14, false)]  // "{\"n\u6F22\u5B57ame\":\"A\u6F22\u5B57hso"
+        [InlineData("-123456789", 1, 0, false)] // "-"
+        [InlineData("0.5", 2, 0, false)]    // "0."
+        [InlineData("10.5e+3", 5, 0, false)] // "10.5e"
+        [InlineData("10.5e-1", 6, 0, false)]    // "10.5e-"
+        [InlineData("{\"i\u6F22\u5B57nts\":[1, 2, 3, 4, 5]}", 27, 25, false)]    // "{\"i\u6F22\u5B57nts\":[1, 2, 3, 4, "
+        [InlineData("{\"s\u6F22\u5B57trings\":[\"a\u6F22\u5B57bc\", \"def\"], \"ints\":[1, 2, 3, 4, 5]}", 36, 36, false)]  // "{\"s\u6F22\u5B57trings\":[\"a\u6F22\u5B57bc\", \"def\""
+        [InlineData("{\"a\u6F22\u5B57ge\":30, \"name\":\"test}:[]\", \"another \u6F22\u5B57string\" : \"tests\"}", 25, 24, false)]   // "{\"a\u6F22\u5B57ge\":30, \"name\":\"test}"
+        [InlineData("   [[[[{\r\n\"t\u6F22\u5B57emp1\":[[[[{\"t\u6F22\u5B57emp2:[]}]]]]}]]]]\":[]}]]]]}]]]]   ", 54, 29, false)] // "   [[[[{\r\n\"t\u6F22\u5B57emp1\":[[[[{\"t\u6F22\u5B57emp2:[]}]]]]}]]]]"
+        [InlineData("{\r\n\"is\u6F22\u5B57Active\": false, \"in\u6F22\u5B57valid\"\r\n : \"now its \u6F22\u5B57valid\"}", 26, 26, false)]  // "{\r\n\"is\u6F22\u5B57Active\": false, \"in\u6F22\u5B57valid\"\r\n}"
+        [InlineData("{\"property\\u1234Name\": \"String value with hex: \\uABCD in the middle.\"}", 51, 23, false)]  // "{\"property\\u1234Name\": \"String value with hex: \\uAB"
+        [InlineData("{ \"number\": 0}", 13, 12, false)]    // "{ \"number\": 0"
+
+        [InlineData("\"a\u6F22\u5B57lpha\" \r\n//This is a comment with a line break\n//Another single-line comment", 53, 16, true)]
+        [InlineData("\"a\u6F22\u5B57lpha\" \r\n//This is a comment with a line break\n//Another single-line comment", 54, 54, true)]
+        [InlineData("\"a\u6F22\u5B57lpha\" \r\n//This is a comment with a carriage return\r\n//Another single-line comment", 59, 16, true)]
+        [InlineData("\"a\u6F22\u5B57lpha\" \r\n//This is a comment with a carriage return\r\n//Another single-line comment", 60, 60, true)]
+        [InlineData("//This is a comment with a carriage return\r//Another single-line comment", 42, 0, true)]
+        [InlineData("//This is a comment with a carriage return\r//Another single-line comment", 43, 0, true)]
+        [InlineData("//This is a comment with a carriage return\r//Another single-line comment", 44, 43, true)]
+        public static void PartialJson(string jsonString, int splitLocation, int consumed, bool containsComments)
         {
             byte[] dataUtf8 = Encoding.UTF8.GetBytes(jsonString);
-
             foreach (JsonCommentHandling commentHandling in Enum.GetValues(typeof(JsonCommentHandling)))
             {
+                if (commentHandling == JsonCommentHandling.Disallow && containsComments)
```

Comment:
Otherwise this test will fail for any partial json that contains a comment

Reply:
Instead of adding this check here, I would add the test cases including comments to the other tests which explicitly support comments:
https://github.com/dotnet/corefx/blob/4378851bc5fd0e3c5043a6d0a3ac835e838fcf7d/src/System.Text.Json/tests/Utf8JsonReaderTests.cs#L965
https://github.com/dotnet/corefx/blob/4378851bc5fd0e3c5043a6d0a3ac835e838fcf7d/src/System.Text.Json/tests/Utf8JsonReaderTests.cs#L1140